src/Tools/dotnet-watch/src/Internal/FileWatcher/PollingFileWatcher.cs

```diff
@@ -188,7 +188,17 @@ private void ForeachEntityInDirectory(DirectoryInfo dirInfo, Action<FileSystemIn
                 return;
             }
 
-            var entities = dirInfo.EnumerateFileSystemInfos("*.*");
+            IEnumerable<FileSystemInfo> entities;
+            try
+            {
+                entities = dirInfo.EnumerateFileSystemInfos("*.*");
+            }
+            // If the directory is deleted after the exists check this will throw and could crash the process
+            catch (DirectoryNotFoundException)
+            {
+                return;
```

Comment:
Do you think the `ForeachEntityInDirectory` needs a similar treatment vis-a-vis files being deleted?

Reply:
? This is in ForeachEntityInDirectory

Reply:
Ahh, I meant for this: https://github.com/dotnet/aspnetcore/pull/21544/files#diff-f043a2280759c2b01c8f9cec3cbc060dR204?

Reply:
I don't think so? It has a catch for `FileNotFoundException` and is using the already resolved `FileSystemInfo` object, so should be pretty safe.